provider.go

```diff
@@ -77,6 +83,63 @@ func readJWTToken(path string) (string, error) {
 	return string(bytes.TrimSpace(data)), nil
 }
 
+func splitSecretPath(secretPath string) (string, string) {
+	s := regexp.MustCompile("/+").Split(secretPath, 3)
+	return s[1], s[2]
```

Comment:
`len(s)` could possibly be less than 3. Given the small size of this function and its single use, it may be appropriate to move the `regexp` call to the calling site and check the result length there (since you'll be able to return an error).